src/mo_frontend/typing.ml

```diff
@@ -2337,7 +2338,12 @@ and infer_dec env dec : T.typ =
   let t =
   match dec.it with
   | ExpD exp
-  | LetD (_, exp) ->
+  | LetD (_, exp, None) -> infer_exp env exp
+  | LetD (_, exp, Some fail) ->
+    if env.in_actor then
+      error env dec.at "M0184" "let-else binding forbidden for actor fields";
```

Comment:
NIT: I wonder if you need to do this check/add this restriction.

I don't think you can `return` or `throw` here anyway, but why shouldn't a user decide to trap with a better message:

```
actor a {
     ...
     let ?b = find(id) else Debug.trap("coudn't find" # id);
     ...
}
```

Reply:
Just talked about this with Gabor. I think the reason to keep the restriction is that:

If you don't restrict this behavior, you can infinitely loop in the else block. And that would be both annoying and expensive to debug.

Also, if we add this restriction now and remove it later (because some user really wants to add custom trap messages in actor field initialization) that's not a breaking change, verses the reverse would be.

What do you think?

Reply:
Hmmm, but this argument also applies to non-actors, like objects...

Reply:
Okay I'll try and remove the restriction